tests/functional/test_config.py

```diff
@@ -0,0 +1,488 @@
+    assert_config_type(*nr_wd_cv, False, True, 'correct_value')
+
+
+def test_not_required_with_default_sent_empty_value(nr_wd_ev):
+    assert_config_type(*nr_wd_ev, False, True, 'empty_value')
+
+
+def test_not_required_with_default_sent_null_value(nr_wd_nv):
+    assert_config_type(*nr_wd_nv, False, True, 'null_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_correct_value', scope='module')
+def nr_wod_cv(sdk_client_ms: ADCMClient, request):
+    return processing_data(
+        sdk_client_ms, request, 'not_required_without_default_sent_correct_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_empty_value', scope='module')
+def nr_wod_ev(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_without_default_sent_empty_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'not_required', 'without_default', 'sent_null_value', scope='module')
+def nr_wod_nv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'not_required_without_default_sent_null_value')
+
+
+def test_not_required_without_default_sent_correct_value(nr_wod_cv):
+    assert_config_type(*nr_wod_cv, False, False, 'correct_value')
+
+
+def test_not_required_without_default_sent_empty_value(nr_wod_ev):
+    assert_config_type(*nr_wod_ev, False, False, 'empty_value')
+
+
+def test_not_required_without_default_sent_null_value(nr_wod_nv):
+    assert_config_type(*nr_wod_nv, False, False, 'null_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'required', 'with_default', 'sent_correct_value', scope='module')
+def r_wd_cv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'required_with_default_sent_correct_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'required', 'with_default', 'sent_empty_value', scope='module')
+def r_wd_ev(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'required_with_default_sent_empty_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'required', 'with_default', 'sent_null_value', scope='module')
+def r_wd_nv(sdk_client_ms: ADCMClient, request):
+    return processing_data(sdk_client_ms, request, 'required_with_default_sent_null_value')
+
+
+def test_required_with_default_sent_correct_value(r_wd_cv):
+    assert_config_type(*r_wd_cv, True, True, 'correct_value')
+
+
+def test_required_with_default_sent_empty_value(r_wd_ev):
+    assert_config_type(*r_wd_ev, True, True, 'empty_value')
+
+
+def test_required_with_default_sent_null_value(r_wd_nv):
+    assert_config_type(*r_wd_nv, True, True, 'null_value')
+
+
+@fixture_parametrized_by_data_subdirs(
+    __file__, 'required', 'without_default', 'sent_correct_value', scope='module')
+def r_wod_cv(sdk_client_ms: ADCMClient, request):
```

Comment:
help message needed. Func name too short to understand what is going on here.

Reply:
name is formed from parameters "fixture_parametrized_by_data_subdirs(
    __file__, 'required', 'without_default', 'sent_correct_value', scope='module')" 
'required', 'without_default', 'sent_correct_value' - r_wod_cv